dipy/io/statefull_tractogram.py

```diff
@@ -0,0 +1,879 @@
+            tmp_affine[0:3, 0:3] = self._affine[0:3, 0:3]
+            shift = apply_affine(tmp_affine, shift)
+        if self._shifted_origin:
+            shift *= -1
+
+        self._streamlines._data += shift
+        if not self._shifted_origin:
+            logging.info('Origin moved to the center of voxel')
+        else:
+            logging.info('Origin moved to the corner of voxel')
+
+        self._shifted_origin = not self._shifted_origin
+
+
+def save_tractogram(sft, filename, bbox_valid_check=True):
+    """ Save the statefull tractogram in any format (trk, tck, fib, dpy)
+
+    Parameters
+    ----------
+    sft : StateFullTractogram
+        The tractogram to save (must have been generated properly)
+    filename : string
+        Filename with valid extension
+
+    Returns
+    -------
+    output : bool
+        Did the saving work properly
+    """
+
+    _, extension = robust_split_name(filename)
+    if extension not in ['.trk', '.tck', '.vtk', '.fib', '.dpy']:
+        TypeError('Output filename is not one of the supported format')
+
+    if bbox_valid_check and not sft.is_bbox_in_vox_valid():
+        raise ValueError('Bounding box is not valid in voxel space, cannot ' +
+                         'save a valid file if some coordinates are invalid')
+
+    old_space = deepcopy(sft.get_current_space())
+    old_shift = deepcopy(sft.get_current_shift())
+
+    sft.to_rasmm()
+    sft.to_center()
+
+    timer = time.time()
+    if extension in ['.trk', '.tck']:
+        tractogram_type = detect_format(filename)
+        header = create_tractogram_header(tractogram_type,
+                                          *sft.get_space_attribute())
+        new_tractogram = Tractogram(sft.get_streamlines(),
+                                    affine_to_rasmm=np.eye(4))
+
+        if extension == '.trk':
+            sft.convert_data_per_point_for_trk()
+            new_tractogram.data_per_point = sft.get_data_per_point()
+            new_tractogram.data_per_streamline = sft.get_data_per_streamline()
+
+        fileobj = tractogram_type(new_tractogram, header=header)
+        nib.streamlines.save(fileobj, filename)
+
+    elif extension in ['.vtk', '.fib']:
+        save_vtk_streamlines(sft.get_streamlines(), filename, binary=True)
+    elif extension in ['.dpy']:
+        dpy_obj = Dpy(filename, mode='w')
+        dpy_obj.write_tracks(sft.get_streamlines())
+
+    logging.debug('Save %s with %s streamlines in %s seconds',
+                  filename, len(sft), round(time.time() - timer, 3))
+
+    if old_space == Space.VOX:
+        sft.to_vox()
+    elif old_space == Space.VOXMM:
+        sft.to_voxmm()
+
+    if old_shift:
+        sft.to_corner()
+
+    return True
+
+
+def load_tractogram(filename, reference, to_space=Space.RASMM,
+                    shifted_origin=False, bbox_valid_check=True,
+                    trk_header_check=True):
+    """ Applies median filter multiple times on input data.
```

Comment:
It does what now? I think this first line of the docstring must be a copy-paste error (?)